### ISSUE TITLE
Bug: [@typescript-eslint/no-unnecessary-type-constraint] false positive on `extends unknown`

### ISSUE THREAD

**Issue by simPod** [typescript-eslint/typescript-eslint#6070]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=JYOwLgpgTgZghgYwgAgJIB4AqyIA9IgAmAzsgK4gDWIA9gO4gB8yA3sgL5A&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWmQ5Nk+aEOb50URNGgdokcGAC+IXUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
interface I<T extends unknown> {}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unnecessary-type-constraint": "error"
  },
};
```

### Expected Result

No error, extending unknown is valuable and is not a default type of `extends` as `any`.

### Actual Result

`Constraining the generic type `T` to `unknown` does nothing and is unnecessary.` which is false.

### Additional Info

https://stackoverflow.com/a/69287624/519333

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `TypeScript`                           | `4.9.3` |
| `ESLint`                               | `8.15.0` |
| `node`                                 | `X.Y.Z` |

**Comment by simPod** [comment]
Hm, I was playing with my code and the rule is triggered on both.

```ts
interface I<T extends unknown> { }
interface I<T extends any> { }
```

I was not aware the `unknown` is default but then `any` is not unnecessary, is it? One of those cases should be valid extend 🤔 

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=JYOwLgpgTgZghgYwgAgJIB4AqyIA9IgAmAzsgK4gDWIA9gO4gB8yA3sgL4BQoksiKGbHgIlkcEAE9mbdkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWmQ5Nk+aEOb50URNGgdokcGAC+IXUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

**Comment by Josh-Cena** [comment]
Both of them mean exactly the same thing. There's no case where you can instantiate one generic with a type but not the other, so they are both redundant.

**Comment by simPod** [comment]
Understand, thanks!
